plugins/survey-resources/src/components/EditQuestion.svelte

```diff
@@ -0,0 +1,462 @@
+<!--
+//
+// Copyright © 2024 Hardcore Engineering Inc.
+//
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+-->
+<script lang="ts">
+  import { MessageBox, getClient } from '@hcengineering/presentation'
+  import { Question, QuestionKind, Survey } from '@hcengineering/survey'
+  import { Button, EditBox, IconDelete, SelectPopup, eventToHTMLElement, showPopup } from '@hcengineering/ui'
+  import { createEventDispatcher } from 'svelte'
+  import survey from '../plugin'
+
+  const client = getClient()
+  const dispatch = createEventDispatcher()
+
+  export let parent: Survey
+  export let index: number
+  export let readonly: boolean = false
+
+  $: question = parent?.questions?.[index] as Question
+  $: options = question?.options ?? []
+  $: questionIcon =
+    question === undefined
+      ? undefined
+      : question.kind === QuestionKind.OPTIONS
+        ? survey.icon.QuestionKindOptions
+        : question.kind === QuestionKind.OPTION
+          ? survey.icon.QuestionKindOption
+          : survey.icon.QuestionKindString
+
+  let newOption = ''
+  let newQuestion = ''
+
+  async function updateParent (): Promise<void> {
```

Comment:
Minor: with parent being updated on every change it feels it'd be more correct to have questions as a collection of separate docs attached to a survey. Then you will be CRUD-ing individual docs (questions).
Can be updated later if needed.

Reply:
I think surveys are not supposed to have a lot of questions. So it was easier to store all them in one place. We can extract them into separate objects if see the performance suffers.